### PR TITLE
fix(mergers): be a tad smarter re $ref resolving

### PR DESCRIPTION
**Pull request by P0lip**
image

### CODE REVIEW

File: src/mergers/mergeAllOf.ts [hunk 1]
```diff
@@ -40,7 +40,8 @@ function _mergeAllOf(fragment: SchemaFragment, path: string[], resolveRef: Walke
 
             if (Array.isArray(resolved.allOf)) {
               for (const member of resolved.allOf) {
-                if (typeof member.$ref === 'string' && schemaRefs.includes(member.$ref)) {
+                const index = schemaRefs.indexOf(member.$ref);
+                if (typeof member.$ref === 'string' && index !== -1 && index !== schemaRefs.lastIndexOf(member.$ref)) {
```
Comment (P0lip) [inline src/mergers/mergeAllOf.ts:44]: I still don't quite like it, but at the same time, we'd need a more robust solution inside of json-schema-merge-allof itself to make come up with something considerably better.

**Review by marbemac — APPROVED** [review]
💯

File: src/accessors/getValidations.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import type { SchemaNodeKind } from '../nodes/types';
 import type { SchemaFragment } from '../types';
 import { pick } from '../utils/pick';
 
-export const COMMON_VALIDATION_TYPES: string[] = ['readOnly', 'writeOnly'];
+export const COMMON_VALIDATION_TYPES: string[] = ['readOnly', 'writeOnly', 'style'];
```
Comment (marbemac) [inline src/accessors/getValidations.ts:7]: I added this little guy - hope it's OK. This allows us to completely ditch the custom parameter rendering we have in elements (only thing missing was tracking the `style` option that http parameters sometimes use).

Comment (P0lip) [reply src/accessors/getValidations.ts:7]: yeah, fair enough.

**Pull request merged**